packages/studio/src/components/PageSettingsButton/EntityPageModal.tsx

```diff
@@ -64,9 +77,33 @@ export default function EntityPageModal({
         updateGetPathValue(pageName, getPathValue);
       }
       updateStreamScope(pageName, StreamScopeParser.parseStreamScope(form));
+      const regenerateTestData = async () => {
+        try {
+          const mapping = await generateTestData();
+          updateEntityFiles(pageName, mapping[pageName]);
+          setActiveEntityFile(mapping[pageName]?.[0]);
+          await refreshActivePageEntities();
+        } catch {
+          toast.warn(
+            "Error generating test data, but entity page settings were still updated."
+          );
+        }
+      };
+      if (form.entityTypes || form.entityIds || form.savedFilterIds) {
+        await regenerateTestData();
```

Comment:
wouldn't we want to generate test data any time even if the stream scope isn't filtered to anything?

Reply:
I updated the guarding logic it wasn't what I intended